src/providers/Guardian.js

```diff
@@ -0,0 +1,86 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import { encodeCallScript } from '@1hive/connect-react'
+import { useContract } from '@/hooks/shared/useContract'
+import tokenManagerAbi from '@/abi/tokenManager.json'
+import { useWallet } from './Wallet'
+import { utils } from 'ethers'
+
+let GUARDIANS_TOKEN_MANAGER = process.env.GUARDIANS_TOKEN_MANAGER
+
+if (!GUARDIANS_TOKEN_MANAGER) {
+  throw new Error('Missing GUARDIANS_TOKEN_MANAGER env variable')
+}
+
+const TOKEN_MANAGER_INTERFACE = new utils.Interface(tokenManagerAbi)
+
+const GuardianContext = React.createContext()
+
+function GuardianProvider({ children }) {
+  const wallet = useWallet()
+  const { account } = wallet
+  const [loading, setLoading] = useState()
+  const [isGuardian, setIsGuardian] = useState(false)
+  const guardiansTokenManager = useContract(
+    GUARDIANS_TOKEN_MANAGER,
+    tokenManagerAbi
+  )
+
+  useEffect(() => {
+    if (!guardiansTokenManager || !account) {
+      return
+    }
+
+    async function checkIfIsGuardian() {
+      setLoading(true)
+      /**
+       * Pass an empty evm script as token manager doesn't use
+       * it to perform a canForward check
+       */
+      try {
+        const canForward = await guardiansTokenManager.canForward(account, '0x')
```

Comment:
Yes, we can do it that way as well.

The reasoning behind using `canForward()` was to avoid doing two calls: one to get the token and other one to get the balance. `canForward` checks the balance already

Reply:
oh, so no problem lets keep it as it is